lib/src/iterable_extensions.dart

```diff
@@ -352,6 +352,11 @@ extension IterableExtension<T> on Iterable<T> {
     }
     return null;
   }
+  
+  /// Returns an element at the given [index]
+  /// or `null` if the [index] is out of bounds of this list.
+  /// [index] must not be negative
+  T? elementAtOrNull(int index) => skip(index).firstOrNull;
```

Comment:
That's awesome code reuse. Probably reasonable performance, otherwise we can optimize.